features/admin.actions.v1/components/pre-update-password-action-config-form.tsx

```diff
@@ -316,114 +352,128 @@ const PreUpdatePasswordActionConfigForm: FunctionComponent<PreUpdatePasswordActi
                     actionTypeApiPath={ actionTypeApiPath }
                     actionId={ initialValues?.id }
                 />
+                { (RuleExpressionsMetaData && showRuleComponent) && (
+                    <RuleConfigForm
+                        readonly={ getFieldDisabledStatus() }
+                        rule={ rule }
+                        setRule={ setRule }
+                        isHasRule={ isHasRule }
+                        setIsHasRule={ setIsHasRule }
+                    />
+                ) }
             </>
         );
     };
 
     return (
-        <FinalForm
-            onSubmit={ (values: PreUpdatePasswordActionConfigFormPropertyInterface, form: any) => {
-                handleSubmit(values, form.getState().dirtyFields);
-            }
-            }
-            validate={ validateForm }
-            initialValues={ initialValues }
-            render={ ({ handleSubmit, form }: FormRenderProps) => (
-                <form onSubmit={ handleSubmit }>
-                    <EmphasizedSegment
-                        className="form-wrapper"
-                        padded={ "very" }
-                        data-componentid={ `${_componentId}-section` }
-                    >
-                        <div className="form-container with-max-width">
-                            { renderFormFields() }
-                            { !isLoading && (
-                                <Button
-                                    size="medium"
-                                    variant="contained"
-                                    onClick={ handleSubmit }
-                                    className={ "button-container" }
-                                    data-componentid={ `${_componentId}-primary-button` }
-                                    loading={ isSubmitting }
-                                    disabled={ getFieldDisabledStatus() }
-                                >
-                                    {
-                                        isCreateFormState
-                                            ? t("actions:buttons.create")
-                                            : t("actions:buttons.update")
+        <RulesProvider
+            conditionExpressionsMetaData={ RuleExpressionsMetaData }
+            initialData={ initialValues?.rule }
+        >
+            <FinalForm
+                onSubmit={ (values: PreUpdatePasswordActionConfigFormPropertyInterface, form: any) => {
```

Comment:
`import type { FormApi } from "final-form";`

```suggestion
                onSubmit={ (values: PreUpdatePasswordActionConfigFormPropertyInterface, form: FormApi<PreUpdatePasswordActionConfigFormPropertyInterface>) => {
```

ref: https://github.com/wso2/identity-apps/blob/master/docs/write-code/CODE_QUALITY.md#refrain-from-using-any-as-a-type-use-it-as-the-last-resort

Reply:
There is a type incompatibility when getting dirtyFields from the form after applying this change. Hence will keep any at this moment